gateway/transport/plugins/jira/jira.go

```diff
@@ -0,0 +1,173 @@
+package jira
+
+import (
+	"encoding/base64"
+	"fmt"
+	"libhoop/log"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	pb "github.com/hoophq/hoop/common/proto"
+	"github.com/hoophq/hoop/gateway/jira"
+	pgreview "github.com/hoophq/hoop/gateway/pgrest/review"
+	pgsession "github.com/hoophq/hoop/gateway/pgrest/session"
+	sessionstorage "github.com/hoophq/hoop/gateway/storagev2/session"
+	"github.com/hoophq/hoop/gateway/storagev2/types"
+	plugintypes "github.com/hoophq/hoop/gateway/transport/plugins/types"
+)
+
+type plugin struct{}
+
+type sessionParseOption struct {
+	withLineBreak bool
+	events        []string
+}
+
+func parseSessionToFile(s *types.Session, opts sessionParseOption) []byte {
+	output := []byte{}
+	for _, eventList := range s.EventStream {
+		event := eventList.(types.SessionEventStream)
+		eventType, _ := event[1].(string)
+		eventData, _ := base64.StdEncoding.DecodeString(event[2].(string))
+
+		if !slices.Contains(opts.events, eventType) {
+			continue
+		}
+
+		switch eventType {
+		case "i":
+			output = append(output, eventData...)
+		case "o", "e":
+			output = append(output, eventData...)
+		}
+		if opts.withLineBreak {
+			output = append(output, '\n')
+		}
+	}
+	return output
+}
+
+func New() *plugin { return &plugin{} }
+
+func (r *plugin) Name() string                            { return plugintypes.PluginJiraName }
+func (r *plugin) OnStartup(ctx plugintypes.Context) error { return nil }
```

Comment:
It's important to run the OnConnect and OnDisconnect phases if the user configured the Jira Integration, otherwise these methods shouldn't run.